<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="8" passed="3" failed="4" skipped="1">
  <reporter-output>
  </reporter-output>
  <suite started-at="2023-09-05T00:53:55 IST" name="Default suite" finished-at="2023-09-05T00:56:04 IST" duration-ms="129798">
    <groups>
    </groups>
    <test started-at="2023-09-05T00:53:55 IST" name="Default test" finished-at="2023-09-05T00:56:04 IST" duration-ms="129798">
      <class name="com.HRM.tests.LoginTest">
        <test-method is-config="true" signature="setup()[pri:0, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:53:55 IST" name="setup" finished-at="2023-09-05T00:54:18 IST" duration-ms="23361" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="validatetitletest()[pri:1, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:54:18 IST" name="validatetitletest" finished-at="2023-09-05T00:54:18 IST" duration-ms="101" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- validatetitletest -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:54:18 IST" name="teardown" finished-at="2023-09-05T00:54:20 IST" duration-ms="1174" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:54:20 IST" name="setup" finished-at="2023-09-05T00:54:30 IST" duration-ms="10055" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="validatelogotest()[pri:2, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:54:30 IST" name="validatelogotest" finished-at="2023-09-05T00:54:33 IST" duration-ms="3132" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- validatelogotest -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:54:33 IST" name="teardown" finished-at="2023-09-05T00:54:35 IST" duration-ms="1972" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:54:35 IST" name="setup" finished-at="2023-09-05T00:54:45 IST" duration-ms="10276" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="logintest()[pri:3, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:54:45 IST" name="logintest" finished-at="2023-09-05T00:54:49 IST" duration-ms="3721" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- logintest -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:com.HRM.tests.LoginTest@4c60d6e9]" started-at="2023-09-05T00:54:49 IST" name="teardown" finished-at="2023-09-05T00:54:50 IST" duration-ms="1086" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
      </class> <!-- com.HRM.tests.LoginTest -->
      <class name="com.HRM.tests.Homepagetest">
        <test-method is-config="true" signature="setup()[pri:0, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:54:50 IST" name="setup" finished-at="2023-09-05T00:55:04 IST" duration-ms="14347" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="validatetitletest()[pri:1, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:04 IST" name="validatetitletest" finished-at="2023-09-05T00:55:04 IST" duration-ms="11" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "com.HRM.Pages.Loginpage.validatetitle()" because "this.log" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "com.HRM.Pages.Loginpage.validatetitle()" because "this.log" is null
at com.HRM.tests.LoginTest.validatetitletest(LoginTest.java:30)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)
at org.testng.SuiteRunner.run(SuiteRunner.java:336)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1200)
at org.testng.TestNG.runSuites(TestNG.java:1114)
at org.testng.TestNG.run(TestNG.java:1082)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- validatetitletest -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:04 IST" name="teardown" finished-at="2023-09-05T00:55:06 IST" duration-ms="1196" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:06 IST" name="setup" finished-at="2023-09-05T00:55:20 IST" duration-ms="14556" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="validatelogotest()[pri:2, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:20 IST" name="validatelogotest" finished-at="2023-09-05T00:55:20 IST" duration-ms="50" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "com.HRM.Pages.Loginpage.validateloge()" because "this.log" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "com.HRM.Pages.Loginpage.validateloge()" because "this.log" is null
at com.HRM.tests.LoginTest.validatelogotest(LoginTest.java:37)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)
at org.testng.SuiteRunner.run(SuiteRunner.java:336)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1200)
at org.testng.TestNG.runSuites(TestNG.java:1114)
at org.testng.TestNG.run(TestNG.java:1082)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- validatelogotest -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:20 IST" name="teardown" finished-at="2023-09-05T00:55:22 IST" duration-ms="1867" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method is-config="true" signature="setup()[pri:0, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:22 IST" name="setup" finished-at="2023-09-05T00:55:38 IST" duration-ms="15360" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="logintest()[pri:3, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:38 IST" name="logintest" finished-at="2023-09-05T00:55:38 IST" duration-ms="29" status="FAIL">
          <exception class="java.lang.NullPointerException">
            <message>
              <![CDATA[Cannot invoke "com.HRM.Pages.Loginpage.validatelogin(String, String)" because "this.log" is null]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.NullPointerException: Cannot invoke "com.HRM.Pages.Loginpage.validatelogin(String, String)" because "this.log" is null
at com.HRM.tests.LoginTest.logintest(LoginTest.java:44)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)
at org.testng.SuiteRunner.run(SuiteRunner.java:336)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1200)
at org.testng.TestNG.runSuites(TestNG.java:1114)
at org.testng.TestNG.run(TestNG.java:1082)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.NullPointerException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- logintest -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:38 IST" name="teardown" finished-at="2023-09-05T00:55:39 IST" duration-ms="1027" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
        <test-method signature="validatehomepagelogo()[pri:4, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:39 IST" name="validatehomepagelogo" finished-at="2023-09-05T00:55:39 IST" depends-on-methods="com.HRM.tests.LoginTest.validatelogotest" duration-ms="0" status="SKIP">
          <exception class="java.lang.Throwable">
            <message>
              <![CDATA[Method com.HRM.tests.Homepagetest.validatehomepagelogo() on instance com.HRM.tests.Homepagetest@70ed52de depends on not successfully finished methods 
[com.HRM.tests.LoginTest.validatelogotest() on instance com.HRM.tests.Homepagetest@70ed52de
com.HRM.tests.LoginTest.validatelogotest() on instance com.HRM.tests.LoginTest@4c60d6e9]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.Throwable: Method com.HRM.tests.Homepagetest.validatehomepagelogo() on instance com.HRM.tests.Homepagetest@70ed52de depends on not successfully finished methods 
[com.HRM.tests.LoginTest.validatelogotest() on instance com.HRM.tests.Homepagetest@70ed52de
com.HRM.tests.LoginTest.validatelogotest() on instance com.HRM.tests.LoginTest@4c60d6e9]
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:156)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)
at org.testng.SuiteRunner.run(SuiteRunner.java:336)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1200)
at org.testng.TestNG.runSuites(TestNG.java:1114)
at org.testng.TestNG.run(TestNG.java:1082)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.Throwable -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- validatehomepagelogo -->
        <test-method is-config="true" signature="setup()[pri:0, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:39 IST" name="setup" finished-at="2023-09-05T00:55:52 IST" duration-ms="13180" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setup -->
        <test-method signature="adminlinktest()[pri:5, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:55:52 IST" name="adminlinktest" finished-at="2023-09-05T00:56:03 IST" duration-ms="11470" status="FAIL">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[expected [true] but found [false]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: expected [true] but found [false]
at org.testng.Assert.fail(Assert.java:111)
at org.testng.Assert.failNotEquals(Assert.java:1578)
at org.testng.Assert.assertTrue(Assert.java:57)
at org.testng.Assert.assertTrue(Assert.java:67)
at com.HRM.tests.Homepagetest.adminlinktest(Homepagetest.java:39)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:568)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:664)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:227)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:957)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:200)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
at org.testng.TestRunner.privateRun(TestRunner.java:848)
at org.testng.TestRunner.run(TestRunner.java:621)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:443)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:437)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:397)
at org.testng.SuiteRunner.run(SuiteRunner.java:336)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1280)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1200)
at org.testng.TestNG.runSuites(TestNG.java:1114)
at org.testng.TestNG.run(TestNG.java:1082)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- adminlinktest -->
        <test-method is-config="true" signature="teardown()[pri:0, instance:com.HRM.tests.Homepagetest@70ed52de]" started-at="2023-09-05T00:56:03 IST" name="teardown" finished-at="2023-09-05T00:56:04 IST" duration-ms="902" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- teardown -->
      </class> <!-- com.HRM.tests.Homepagetest -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
